src/gfn/gym/hypergrid.py

```diff
@@ -210,6 +225,106 @@ def reward(self, states: DiscreteStates) -> torch.Tensor:
         ), f"reward.shape is {reward.shape} and states.batch_shape is {states.batch_shape}"
         return reward
 
+    # -------------------------
+    # Mode utilities
+    # -------------------------
+    def _mode_reward_threshold(self) -> float:
```

Comment:
this method could be moved inside GridReward object and use polymorphism instead of the if chain